twisted/internet/_sslverify.py

```diff
@@ -1942,3 +1901,61 @@ def fromFile(cls, filePath):
             <twisted.internet.ssl.DiffieHellmanParameters>}
         """
         return cls(filePath)
+
+
+def _setUpNextProtocolMechanisms(context, acceptableProtocols):
```

Comment:
I have no particular opinions regarding the naming. As I recall, the last time we did an NPN/ALPN patch we also spent several review cycles on the naming. This is in large part because "protocol" is *already* an overloaded term in Twisted land, and so adding another possible meaning to it does not improve clarity, no matter how we spin it. =P